### PR TITLE
Use the last part of the path when trying to find matching process.

### LINKED ISSUE #895: Launcher fails to find default process when process type contains `.`
### Summary

This was initially uncovered in https://github.com/paketo-buildpacks/dotnet-publish/pull/356, where the Paketo buildpacks use .NET Core project names as process types. Some of these look like `Steeltoe.Demo`. It's common for .NET project names to contain `.`. The buildpacks spec [states](https://github.com/buildpacks/spec/blob/main/buildpack.md#launchtoml-toml) that `.` is a valid character in a process type name. However, when process types contain this character, the launcher fails to identify the default process at app launch time. The launcher fails with the error
```
ERROR: failed to launch: determine start command: when there is no default process a command is required
```
---

##### Steps

Follow the reproduction steps in the README.md for this minimal sample buildpack + sample app: https://github.com/fg-j/explorations/tree/main/launcher-bug

##### Current behavior

```
docker run --rm -it sample-app
ERROR: failed to launch: determine start command: when there is no default process a command is required
```

##### Expected

```
docker run --rm -it sample-app
Default entrypoint process
```

---

##### lifecycle version

```
pack builder inspect cnbs/sample-builder:bionic
Inspecting builder: cnbs/sample-builder:bionic

REMOTE:

Created By:
  Name: Pack CLI
  Version: 0.27.0+git-f4f5be1.build-3382

Trusted: No

Stack:
  ID: io.buildpacks.samples.stacks.bionic

Lifecycle:
  Version: 0.14.1
  Buildpack APIs:
    Deprecated: (none)
    Supported: 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8
  Platform APIs:
    Deprecated: (none)
    Supported: 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9

Run Images:
  cnbs/sample-stack-run:bionic
```

##### platform version(s)

```
pack report
Pack:
  Version:  0.27.0+git-f4f5be1.build-3382
  OS/Arch:  darwin/amd64

Default Lifecycle Version:  0.14.1

Supported Platform APIs:  0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9
```

### PR DESCRIPTION
**Pull request by jabrown85**
The process was comparing the base path minus the extension, which was blocking types with a `.` in them from ever matching.

Fixes #895

Signed-off-by: Jesse Brown <jabrown85@gmail.com>

### CODE REVIEW

**Review by samj1912 — DISMISSED** [review]

File: go.sum [hunk 1]
```diff
@@ -26,28 +26,16 @@ cloud.google.com/go v0.84.0/go.mod h1:RazrYuxIK6Kb7YrzzhPoLmCVzl7Sup4NrbKPg8KHSU
 cloud.google.com/go v0.87.0/go.mod h1:TpDYlFy7vuLzZMMZ+B6iRiELaY7z/gJPaqbMx6mlWcY=
 cloud.google.com/go v0.90.0/go.mod h1:kRX0mNRHe0e2rC6oNakvwQqzyDmg57xJ+SZU1eT2aDQ=
 cloud.google.com/go v0.93.3/go.mod h1:8utlLll2EF5XMAV15woO4lSbWQlk8rer9aLOfLh7+YI=
-cloud.google.com/go v0.94.1/go.mod h1:qAlAugsXlC+JWO+Bke5vCtc9ONxjQT3drlTTnAplMW4=
```
Comment (jabrown85) [inline go.sum:29]: This must've come from `make unit` running `tidy`. Not sure which commit originally left this

**Comment by jabrown85** [comment]
I went ahead and added a test to prevent future regressions

**Comment by jabrown85** [comment]
Meh, I dunno what is going on with windows and don't currently have a machine 😢 

It isn't getting an error saying the type isn't found, but it seems to be trying to start `process.with.period` I guess? I dunno, it looks the same as `web` to me. I'll dig in later.
```
    launcher_test.go:85: failed to run [docker run --rm --entrypoint=process.with.period --env=CNB_PLATFORM_API=0.10 lifecycle/acceptance/launcher]
         OUTPUT: docker: Error response from daemon: container 1aa8b8c6d9f74977a59d55e7d70c2b272aae4a1a05026a202442838ae466ace7 encountered an error during hcsshim::System::CreateProcess: failure in a Windows system call: The system cannot find the file specified. (0x2).
        
         ERROR: exit status 127
```

**Comment by natalieparellano** [comment]
As discussed in Slack, will include this in release/0.14.2. It will be easier to first merge this into main and then cherry-pick the commit onto the release branch.

**Review by natalieparellano — APPROVED** [review]

**Pull request merged**